src/main/java/carRacing/RacingGame.java

```diff
@@ -5,13 +5,17 @@
 
 public class RacingGame {
 
-    private List<RacingCar> racingCars = new ArrayList<>();
+    private List<RacingCar> racingCars;
+    private int tryCount;
 
-    public List<RacingCar> initRacingCars(int carCount) {
-        for (int i = 0; i < carCount; i++) {
-            racingCars.add(new RacingCar());
-        }
-        return racingCars;
+    public RacingGame(String name) {
+        racingCars = CarFactory.from(name);
+        tryCount = 0;
+    }
+
+    public RacingGame(String name, int tryCount) {
+        racingCars = CarFactory.from(name);
+        this.tryCount = tryCount;
     }
```

Comment:
```suggestion
    public RacingGame(String name) {
        this(CarFactory.from(name), 0);
    }

    public RacingGame(String name, int tryCount) {
        this(CarFactory.from(name), tryCount);
    }
    
    public RacingGame(List<Car> cars, int tryCount) {
        racingCars = cars;
        this.tryCount = tryCount;
    }
```

객체의 상태 값과 같은 타입을 주 생성자로 추가하고, 필드에 대한 초기화는 주 생성자만 담당한다.
부 생성자를 주 생성자를 호출해 초기화를 담당한다.
2주차 강의 영상 참고